plans/verify/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/ipfs/testground/sdk/runtime"
+	"github.com/ipfs/testground/sdk/sync"
+	"github.com/sparrc/go-ping"
+)
+
+func main() {
+	testcases := map[string]runtime.TestCaseFn{
+		"uses-data-network": UsesDataNetwork,
+	}
+	runtime.InvokeMap(testcases)
+}
+
+func setupNetwork(ctx context.Context, runenv *runtime.RunEnv) (*sync.Client, error) {
+	client := sync.MustBoundClient(ctx, runenv)
+	return client, client.WaitNetworkInitialized(ctx, runenv)
+}
+
+func isControlNet(nw string) bool {
+	return strings.HasPrefix(nw, "192.18.") || strings.HasPrefix(nw, "100.96.")
+}
+
+// UsesDataNetwork verifies that instances can only reach each other through the data network.
+// One instance acts as the target. The target publishes the IP addresses it finds to the sync
+// service. Other instances will subscribe to the topic and test for network connectivity to the
+// target on each of its ip addresses.
+// An error is reported if the target is reachable over the control network or if there is packet
+// loss over the data network.
+func UsesDataNetwork(runenv *runtime.RunEnv) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+
+	client, err := setupNetwork(ctx, runenv)
+	if err != nil {
+		runenv.RecordFailure(err)
+		return err
+	}
+	defer client.Close()
+
+	const (
+		_ int64 = iota
+		targetmode
+		pingmode
+	)
+
+	netTopic := sync.NewTopic("addrs", "")
+
+	switch client.MustSignalAndWait(ctx, "ready", runenv.TestInstanceCount) {
+	case targetmode:
+		runenv.RecordMessage("target mode. publishing target networks.")
+		for _, iname := range []string{"eth0", "eth1"} {
+			iface, err := net.InterfaceByName(iname)
+			if err != nil {
+				runenv.RecordFailure(err)
+				return err
+			}
+			addrs, err := iface.Addrs()
+			if err != nil {
+				runenv.RecordFailure(err)
+				return err
+			}
+			for _, addr := range addrs {
+				runenv.RecordMessage("publishing %s", addr)
+				client.Publish(ctx, netTopic, addr.String())
+			}
+		}
+		client.Publish(ctx, netTopic, ".")
```

Comment:
Please extract `"."` into a var and name it somehow - `endOfStream` or `endOfNetworks` or something like that.